sdk/eventhubs/azure_messaging_eventhubs/src/producer/mod.rs

```diff
@@ -479,9 +479,11 @@ impl ProducerClient {
             let expires_at = token.expires_on;
             cbs.authorize_path(&url, token.token.secret(), expires_at)
                 .await?;
-            scopes
-                .insert(url.clone(), token)
-                .ok_or_else(|| Error::from(ErrorKind::UnableToAddAuthenticationToken))?;
+            let present = scopes.insert(url.clone(), token);
+            // insert returns some if it *fails* to insert, None if it succeeded.
```

Comment:
How very Win32 of them. 🙄

Reply:
Yeah, I was *not* happy to realize that.

In retrospect, it makes sense, but there isn't an invert of `unwrap()` :(.